test/test.dart

```diff
@@ -135,6 +135,9 @@ Future<void> setUpAppTest([VoidCallback? configureFakes]) async {
   binding.defaultBinaryMessenger.setMockMethodCallHandler(const MethodChannel('com.ryanheise.audio_session'), (MethodCall methodCall) async {
     return null;
   });
+  binding.defaultBinaryMessenger.setMockMethodCallHandler(const MethodChannel('com.nt4f04und.android_content_provider/ContentResolver'), (MethodCall methodCall) async {
```

Comment:
Since you are the author of the library, it would be awesome if you could expose the `MethodChannel` for testing, similar to how `SystemChannels.platform` is exposed.

Another idea, which would avoid needing to know the internals of the plugin and the method channel, but requires a bit more work, would be to support setting a mock `AndroidContentResolver` by specifying individual callbacks. Something that would look like this:

```dart
AndroidContentResolver.setTestMock(
  bulkInsert: ({required String uri, required List<ContentValues> values}) => 2,
  delete: ({required String uri, String? selection, List<String>? selectionArgs}) => 1,
)
```

Reply:
> Since you are the author of the library, it would be awesome if you could expose the MethodChannel for testing, similar to how SystemChannels.platform is exposed.

There's practically no difference, so I'd prefer to keep it private, I've also never seen any plugin exposing its channels like this.

Adding mock code to the plugin's production code seems very lame, it would also need use to include a dependency on `flutter_test`, which is extremely bad.

Reply:
> There's practically no difference

The difference is requiring to know less about the internals of the plugin. The method channel name is not part of the public api of your plugin and could therefore change without notice, breaking our tests.

> I've also never seen any plugin exposing its channels like this.

Don't feel like that's a reason not to do it, also see [connectivity_plus](https://github.com/fluttercommunity/plus_plugins/blob/5e4c28f285356d8884c7f320b164b6b13c013ba3/packages/network_info_plus/network_info_plus_platform_interface/lib/method_channel_network_info.dart#L16) and [firebase](https://pub.dev/documentation/firebase_core_platform_interface/latest/firebase_core_platform_interface/MethodChannelFirebase/channel-constant.html).

> it would also requiure us to include a dependency on flutter_test

Why would you need that requirement?

> Adding mock code to the plugin's production code seems very lame

The idea behind it is to create a stable public interface for testing, one that can't just change between versions without notice.

Reply:
Exposed the channel.

> Why would you need that requirement?

To set up a mock of the method channel.

> The idea behind it is to create a stable public interface for testing, one that can't just change between versions without notice.

I'm strongly against adding any test/mock code to part of the plugin, except its own tests.